python/cudf/cudf/core/column/lists.py

```diff
@@ -360,14 +375,45 @@ def get(self, index: int) -> ParentType:
         1    5
         2    6
         dtype: int64
+
+        >>> s.list.get(2)
+        0    <NA>
+        1       5
+        2       6
+        dtype: int64
+
+        >>> s.list.get(2, default=0)
+        0   0
+        1   5
+        2   6
+        dtype: int64
+
+        >>> s.list.get([0, 1, 2])
+        0   1
+        1   4
+        2   6
+        dtype: int64
         """
-        min_col_list_len = self.len().min()
-        if -min_col_list_len <= index < min_col_list_len:
-            return self._return_or_inplace(
-                extract_element(self._column, index)
-            )
+        if is_scalar(index):
+            out = extract_element_scalar(self._column, cudf.Scalar(index))
         else:
-            raise IndexError("list index out of range")
+            index = as_column(index)
+            out = extract_element_column(self._column, as_column(index))
```

Comment:
What happens if `len(index) != len(self)`? Is there a clear error?

At first, I misunderstood what this method would do. I thought passing a sequence of ints would extract multiple values, so that `s.list.get([0, 1])` would return a lists column with the first two elements of each contained list. This is really more like a segmented gather. The docs could be improved by specifying that the input indices correspond to one index per list row.

Reply:
I tried to make the docs more clear about this - could you take a look or suggest improvements?

Reply:
@bdice just in case you had any follow-ups here.

Reply:
I'm happy with the outcome here. Sorry, I forgot to reply on this thread when I approved. :)